### PR TITLE
Add cost to run, advance and purge messages

### PR DESCRIPTION
**Pull request by meatcompute**
Finishes #1302 

screen shot 2016-03-07 at 2 50 39 pm
screen shot 2016-03-07 at 2 50 44 pm
screen shot 2016-03-07 at 3 32 13 pm

### CODE REVIEW

File: src/clj/game/core-actions.clj [hunk 1]
```diff
@@ -262,15 +262,16 @@
...
   [state side {:keys [server] :as args}]
   (when (and (not (get-in @state [:runner :register :cannot-run]))
              (can-run-server? state server)
              (can-pay? state :runner "a run" :click 1))
-    (system-msg state :runner (str "makes a run on " server))
-    (run state side server)
-    (pay state :runner nil :click 1)))
+    (let [cost (pay state :runner nil :click 1)]
+      (system-msg state :runner (str (build-spend-msg cost "make a run on") server))
+      (run state side server))))
```
Comment (Saintis) [inline src/clj/game/core-actions.clj]: Paying cost before triggering the run is likely breaking sundew, as now the click is spent outside a run. Should be fixed by moving the `(run ...)` above the let.

**Comment by Saintis** [comment]
Maybe add a triple-click message to purge as well?

**Comment by meatcompute** [comment]
Oh, nice, glad that Sundew was caught.  Let me return to original ordering.

**Comment by Saintis** [comment]
All good, nice work!

**Comment by meatcompute** [comment]
Tested out cost logging in `do-purge` and it's just as simple as an addition. pushing up a change in a sec then I expect it to be good for merge.

**Comment by JoelCFC25** [comment]
@meatcompute: Would you like to join the developer Slack? If so, send an email to @mtgred. More server side and front end improvements are definitely welcome--great job!

**Comment by meatcompute** [comment]
Sure! And thank you

**Comment by mtgred** [comment]
:+1:

**Pull request merged**